src/Core/Domain/MerchandiseReturn/Command/DeleteProductFromMerchandiseReturnCommand.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\MerchandiseReturn\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\MerchandiseReturn\ValueObject\CustomizationId;
+use PrestaShop\PrestaShop\Core\Domain\MerchandiseReturn\ValueObject\MerchandiseReturnDetailId;
+use PrestaShop\PrestaShop\Core\Domain\MerchandiseReturn\ValueObject\MerchandiseReturnId;
+
+/**
+ * @todo I am not sure about naming, since technically it's detail not a product
+ * Maybe DeleteMerchandiseReturnDetailCommand? and don't use FromMerchandiseReturn
+ * Deletes product from given order.
+ */
+class DeleteProductFromMerchandiseReturnCommand
+{
+    /**
+     * @var MerchandiseReturnId
+     */
+    private $merchandiseReturnId;
+
+    /**
+     * @var MerchandiseReturnDetailId
+     */
+    private $merchandiseReturnDetailId;
+
+    /**
+     * @var CustomizationId|null
+     */
+    private $customizationId;
+
+    /**
+     * DeleteProductFromMerchandiseReturnCommand constructor.
+     *
+     * @param int $merchandiseReturnId
+     * @param int $merchandiseReturnDetailId
+     * @param int $customizationId
+     *
+     * @throws \PrestaShop\PrestaShop\Core\Domain\MerchandiseReturn\Exception\MerchandiseReturnConstraintException
+     */
+    public function __construct(int $merchandiseReturnId, int $merchandiseReturnDetailId, int $customizationId = 0)
+    {
+        $this->merchandiseReturnId = new MerchandiseReturnId($merchandiseReturnId);
+        $this->merchandiseReturnDetailId = new MerchandiseReturnDetailId($merchandiseReturnDetailId);
+        if ($customizationId !== 0) {
```

Comment:
This needs to be properly explained.
A small comment will do 😄 + phpDoc
`@param int $customizationId must be zero if no customization`
```
// if zero is passed, it means there is no customization ID and we keep it as `null`
if ($customizationId !== 0) {
```

I'm wondering 🤔 shall we accept `null` as well ?
`,$customizationId = null)`

Reply:
Yeah I think it makes sense.